addons/website_sale/models/account_move.py

```diff
@@ -8,7 +8,7 @@ class AccountMove(models.Model):
     _inherit = 'account.move'
 
     website_id = fields.Many2one(
-        'website', related='partner_id.website_id', string='Website',
+        'website', related='line_ids.sale_line_ids.order_id.website_id', string='Website',
         help='Website through which this invoice was created for eCommerce orders.',
         store=True, readonly=True, tracking=True)
```

Comment:
Very advanced case, but I suppose this could break if you invoice multiple orders of the same partner from different websites at once :thinking:.
Instead of a related field, shouldn't we use a simple stored field, which is given in the invoice creation values ?

Reply:
I have actually tested that case and In case the invoice has more than one SO associated with, the `website_id` relates to the `website_id` of the **_most recent_** SO. However, I don't think it will lead to a bug as the automatic invoice -to my knowledge- will always relate to one SO. Let me know what you think!

Reply:
Ok then, but I'd still prefer to avoid side-effects, by using a true compute method (with no or nearly no dependencies, since it only matters to compute it on invoice creation), s.t. we limit the potentially useless recomputation and performance costs.
And I'll make a task to change the field in master s.t. the invoice gets the so website directly in its creation values.